src/main/java/gyro/google/dns/DnsPolicyNetwork.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.dns;
+
+import java.util.Optional;
+
+import com.google.api.services.dns.model.PolicyNetwork;
+import gyro.core.resource.Diffable;
+import gyro.core.validation.Required;
+import gyro.google.Copyable;
+import gyro.google.compute.NetworkResource;
+
+public class DnsPolicyNetwork extends Diffable implements Copyable<PolicyNetwork> {
+
+    private NetworkResource network;
+
+    /**
+     * The VPC network to bind to.
+     *
+     * @resource gyro.google.compute.NetworkResource
+     */
+    @Required
+    public NetworkResource getNetwork() {
+        return network;
+    }
+
+    public void setNetwork(NetworkResource network) {
+        this.network = network;
+    }
+
+    @Override
+    public void copyFrom(PolicyNetwork model) {
+        String networkUrl = model.getNetworkUrl();
+
+        if (networkUrl != null) {
+            setNetwork(findById(NetworkResource.class, networkUrl));
+        }
+    }
+
+    @Override
+    public String primaryKey() {
+        return Optional.ofNullable(getNetwork())
+            .map(NetworkResource::primaryKey)
```

Comment:
Because you are using the primary key of the network resource, if the configuration is like
```
google::dns-policy dns-policy-example
    name: "dns-policy-example"
    description: "DNS Policy Example"
    network
        network: $(external-query google::compute-network {name: "vpc-example"}).0.self-link
    end
    network
        network: $(external-query google::compute-network {name: "vpc-example-asung"}).0.self-link
    end
end
```

then an error similar to this occurs after you gyro up, after create
```
$ ../../gyro/cli/dist/gyro up dns/policy.gyro --verbose
↓ Loading plugin: gyro:gyro-google-provider:0.99.1-SNAPSHOT
⟳ Refreshed resources: 1

Unexpected error: java.lang.IllegalArgumentException: Multiple entries with same key: Optional[google::compute-network::null]=network google::compute-network::null and Optional[google::compute-network::null]=network google::compute-network::null
	at com.google.common.collect.ImmutableMap.checkNoConflict(ImmutableMap.java:190)
	at com.google.common.collect.RegularImmutableMap.checkNoConflictInKeyBucket(RegularImmutableMap.java:109)
	at com.google.common.collect.RegularImmutableMap.fromEntryArray(RegularImmutableMap.java:95)
	at com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:373)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:582)
	at gyro.core.scope.NodeEvaluator.lambda$copy$37(NodeEvaluator.java:569)
	at java.base/java.util.Optional.map(Optional.java:265)
	at gyro.core.scope.NodeEvaluator.copy(NodeEvaluator.java:569)
	at gyro.core.scope.NodeEvaluator.lambda$visitResource$35(NodeEvaluator.java:523)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at gyro.core.scope.NodeEvaluator.visitResource(NodeEvaluator.java:523)
	at gyro.core.scope.NodeEvaluator.visitResource(NodeEvaluator.java:81)
	at gyro.lang.ast.block.ResourceNode.accept(ResourceNode.java:55)
	at gyro.lang.ast.NodeVisitor.visit(NodeVisitor.java:37)
	at gyro.core.scope.NodeEvaluator.lambda$evaluateBody$31(NodeEvaluator.java:341)
	at gyro.core.scope.Defer.execute(Defer.java:50)
	at gyro.core.scope.NodeEvaluator.evaluateBody(NodeEvaluator.java:341)
	at gyro.core.scope.NodeEvaluator.visitFile(NodeEvaluator.java:424)
	at gyro.core.scope.NodeEvaluator.visitFile(NodeEvaluator.java:81)
	at gyro.lang.ast.block.FileNode.accept(FileNode.java:32)
	at gyro.lang.ast.NodeVisitor.visit(NodeVisitor.java:37)
	at gyro.core.scope.NodeEvaluator.lambda$evaluateBody$31(NodeEvaluator.java:341)
	at gyro.core.scope.Defer.execute(Defer.java:50)
	at gyro.core.scope.NodeEvaluator.evaluateBody(NodeEvaluator.java:341)
	at gyro.core.scope.NodeEvaluator.evaluate(NodeEvaluator.java:325)
	at gyro.core.scope.RootScope.evaluate(RootScope.java:313)
	at gyro.core.command.AbstractConfigCommand.doExecute(AbstractConfigCommand.java:133)
	at gyro.core.command.AbstractCommand.execute(AbstractCommand.java:71)
	at gyro.cli.Gyro.run(Gyro.java:195)
	at gyro.cli.Gyro.main(Gyro.java:76)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
```

The reason being that the network associated is  not managed by gyro and is external, so the primary key for the Network Resource is null and there are multiple of them which creates this scenario.

Instead using the selflink as the primary key would solve this issue.

All other places where the primary key of a subresource is derived from the primary key of the parent which is a Resource should be modified to use the field which has the `@Id` field